refactor(app-headless-cms): improve menu structure and code clarity

ku5ic opened the pull request:
## Changes

Refactored the Headless CMS admin menu components for better clarity, reusability, and maintainability. Introduced helper components for main menu, content menu, and conditional menu rendering. Updated content groups menu items to use new structure and improved permission handling. This change enhances code readability and keeps the menu logic DRY.

<details>
<summary>Screenshots</summary>
<img width="255" height="497" alt="Screenshot 2025-10-02 at 14 54 37" src="https://github.com/user-attachments/assets/89fbf3d5-55ac-48f8-9095-29e14223c9bf" />
</details>

Ref: [WEB-5362](https://webiny.atlassian.net/browse/WEB-5362)

## How Has This Been Tested?
N/A

## Documentation
N/A

ku5ic committed:
refactor(app-headless-cms): improve menu structure and code clarity

Refactored the Headless CMS admin menu components for better clarity,
reusability, and maintainability. Introduced helper components for main
menu, content menu, and conditional menu rendering. Updated content
groups menu items to use new structure and improved permission handling.
This change enhances code readability and keeps the menu logic DRY.

Ref: WEB-5362

ku5ic committed:
feat(cms): update Content menu icon

Replaced the Content menu icon in the CMS admin with a new icon from
@webiny/icons (wysiwyg.svg) for improved visual clarity and consistency.

Ref: WEB-5362

ku5ic committed:
feat(cms): rename menu item to "Content Modeling"

Renamed the "Headless CMS" menu item to "Content Modeling" in the admin
menu for improved clarity and user understanding.

Ref: WEB-5362

ku5ic committed:
feat(cms): update CMS menu icon to new web icon

Replaced the old Headless CMS menu icon with the new web icon from
@webiny/icons/web.svg for improved consistency and visual clarity in the
admin interface.

Ref: WEB-5362

adrians5j commented on packages/app-headless-cms/src/admin/menus/CmsMenuLoader.tsx:
```diff
@@ -62,30 +120,13 @@ const CmsMenuLoaderComponent = () => {
-                }
-            />
-
+            <CmsMainMenu />
```
I'd maybe call this `CmsContentModelingMenu` but it's a nitpick.

ku5ic replied:
Resolved with [c758c8e](https://github.com/webiny/webiny-js/pull/4725/commits/c758c8eaf5fcccc6048c48c7f87c254847f268e0)

ku5ic replied:
Resolved with 4d090d4c75086100a9b3cb10a0baf23880057f4f

adrians5j commented on packages/app-headless-cms/src/admin/menus/CmsMenuLoader.tsx:
```diff
@@ -62,30 +120,13 @@ const CmsMenuLoaderComponent = () => {
-                    <CmsContentModelsMenu canAccess={canCreateContentModels} />
-                    <CmsContentGroupsMenu canAccess={canCreateContentModelGroups} />
-                </>
+                <CmsCreateMenus
```
Another nitpick, but why even have `CmsCreateMenus`. Why not simply put both components (that are contained in `CmsCreateMenus`) here directly?

Might be easier to read.

ku5ic replied:
Resolved with 4d090d4c75086100a9b3cb10a0baf23880057f4f

adrians5j commented on packages/app-headless-cms/src/admin/menus/ContentGroupsMenuItems.tsx:
```diff
@@ -13,114 +13,134 @@ import { Routes } from "~/routes.js";
-    contentModel?: CmsModel;
-    children: JSX.Element;
+  group: CmsGroup;
+  contentModel?: CmsModel;
```
Hm, why formatting differences? 

Can you run `yarn prettier:fix` locally and see if it does anything?

ku5ic replied:
Resolved with 21b68a37b951cfe469bfeae6a85ed89cc4f993a4

adrians5j commented on packages/app-headless-cms/src/admin/menus/ContentGroupsMenuItems.tsx:
```diff
@@ -13,114 +13,134 @@ import { Routes } from "~/routes.js";
-    children
+  group,
+  contentModel,
+  children,
```
Something is definitely off with formatting, as our rules do not have allow `,` for last item. Weird. 🤔

ku5ic replied:
Resolved with 21b68a37b951cfe469bfeae6a85ed89cc4f993a4

adrians5j reviewed (changes requested):
Looks good. Just a couple of comments. I think the only must-have is formatting. This would get reported by CI, but yeah, since it's not working atm, I had to point it out :)

ku5ic committed:
refactor(cms): rename main menu and inline create menus

Renamed `CmsMainMenu` to `CmsContentModelingMenu` for clarity and updated
usage throughout the file. Inlined the conditional rendering of content
model and group menus, removing the `CmsCreateMenus` component for
simpler structure. This improves menu naming consistency and reduces
component nesting.

ku5ic committed:
feat(cms): refactor ContentGroupsMenuItems formatting

Refactored ContentGroupsMenuItems.tsx to improve code readability and
consistency by updating indentation and formatting. No functional changes
were made. This enhances maintainability and aligns with project style
guidelines.

Ref: WEB-5362

adrians5j commented on packages/app-headless-cms/src/admin/menus/ContentGroupsMenuItems.tsx:
```diff
@@ -67,60 +67,75 @@ export const ContentGroupsMenuItems = () => {
+     * Renders a menu item for a given content group.
+     * Displays the group's name and icon in the menu.
+     */
+    const GroupMenu = ({ group }: { group: CmsGroup }) => (
```
Should we pull this component outside of `ContentGroupsMenuItems`?

This way the component is recreated on each `ContentGroupsMenuItems` component render. I say we do it, unless I'm missing smt.

ku5ic replied:
Resolved with 010e048c2ad21b505b13135bfcc4b11cdc940226

adrians5j commented on packages/app-headless-cms/src/admin/menus/ContentGroupsMenuItems.tsx:
```diff
@@ -67,60 +67,75 @@ export const ContentGroupsMenuItems = () => {
+     * If the group has no content models, displays a "Nothing to show" message.
+     * Wraps each content model menu item with permission checks.
+     */
+    const GroupContentModels = ({ group }: { group: CmsGroup }) => {
```
Same here I'd say. 

Additionally, we could even extract these into separate sibling files (idk, maybe `CmsGroupMenuItem` / `CmsModelMenuItems`), just so we don't have everything in a single file.

ku5ic replied:
Resolved with 010e048c2ad21b505b13135bfcc4b11cdc940226

adrians5j reviewed (changes requested):
Getting closer, thanks!

ku5ic committed:
refactor(cms): split content group menu components

Separated ContentGroupsMenuItems logic into distinct components:
GroupMenu, GroupContentModels, HasContentEntryPermissions, and Icon.
Improves code readability and maintainability by modularizing menu item
rendering and permission checks for CMS content groups.

Ref: WEB-5362

ku5ic committed:
refactor(admin/menus): use named exports for menu components

Switch all menu-related components in the headless CMS admin package to
named exports instead of default exports. This improves consistency,
enables better tree-shaking, and aligns with modern React best
practices. Updated all imports to use named imports accordingly.

ku5ic committed:
fix(cms): simplify content entry permission checks

Refactored HasContentEntryPermissions to streamline permission logic for
content models and groups. Combined conditional checks and removed
redundant branches to improve readability and maintainability.

ku5ic committed:
Merge branch 'next' into feature/WEB-5362/menu_items_refactoring
